devops/poseidon-deployer/tasks/deployPoseidon.ts

```diff
@@ -0,0 +1,96 @@
+.setAction(async ({ multicall, factoryAddress, salt }, hre) => {
+  let poseidonContracts;
+  switch (multicall) {
+    case true:
+      if (factoryAddress == undefined) {
+        throw new Error("failed to invoke multicall due to poseidonFactory address is missing in paramter. Please pass an address using --factory-address (e.g. --factoryAddress 0xabcd..ef");
+      }
+      if (salt == undefined) {
+        throw new Error("failed to invoke multicall due to salt is missing in parameter. Please pass a salt using --salt (e.g. --salt 0x000000000000000000000000000000000000000000000000000000000000000f)");
+      }
+
+      poseidonContracts = await hre.run("deploy-poseidon:callPoseidonFactory", { 
+        factoryAddress: factoryAddress,
+        salt: salt });
+        break;
+
+    default:
+      poseidonContracts = await hre.run("deploy-poseidon:multipleTx");
+  }
+
+  return poseidonContracts;
+});
+
+
+subtask("deploy-poseidon:multipleTx")
+.setAction(async (_, { ethers }) => {
+    let poseidonAddressBook = new Map();
+
+    // Deploy poseidonTN contracts (N: 2, 3, ..., 7)
+    for (let TN = 2; TN <= 7; TN++) {
+      try {
+      const factory = await ethers.getContractFactory(`PoseidonT${TN}`);
+      const contract = await factory.deploy();
+      const address = contract.address;
+
+      console.log(`PoseidonT${TN} deployed: ${address}`);
+
+      poseidonAddressBook.set(`T${TN}`, address);
+    } catch (e) {
+      throw new Error(`failed to deploy contract: PoseidonT${TN} \n` + e);
+    }
+    }
+    
+    return poseidonAddressBook;
+  });
+
+subtask("deploy-poseidon:callPoseidonFactory")
+.addParam("factoryAddress")
+.addParam("salt")
+.setAction(async ({ factoryAddress, salt }, { ethers }) => {
+  const poseidonFactory: Contract = await ethers.getContractAt("PoseidonFactory", factoryAddress);
+  
+  const calls = [];
+  let poseidonAddressBook = new Map();
+  try {
+    for (let TN = 2; TN <= 7; TN++) {
+      // determine each contract addresses
+      const address = await poseidonFactory[`determinePoseidonT${TN}`](salt);
+      poseidonAddressBook.set(`T${TN}`, address);
+
+      // packing contract creation calls into an array
+      const call = poseidonFactory.interface.encodeFunctionData(`deployPoseidonT${TN}`, [ salt ]);
+      calls.push(call);
+    }
+  } catch (e) {
+    throw new Error("failed to call poseidonFactory. Maybe contract address is wrong: \n" + e);
+  }
+
+  try {
+    // deploy contracts
+  const results = await poseidonFactory.multicall(calls);
+  console.log("multicall result: " + JSON.stringify(results));
+  } catch(e){
+    throw new Error('failed to process multicall. Contracts had not been deployed. Maybe salt has been already used to deploy poseidon: \n' + e);
+  }
+  
+  const addresses = Array.from(poseidonAddressBook.values());
+
+  // Ensure deployed contracts are correctly process poseidon functions
```

Comment:
Add what the verification being run is (assuming it's hashing all zeros)

Reply:
Mentioned verification is done by comparing to reference implementation:

https://github.com/privacy-scaling-explorations/maci/blob/3f753a1792931a5910c3614db2426bef2bb134c1/devops/poseidon-deployer/tasks/deployPoseidon.ts#L100-L106